testkit/javadsl/src/main/scala/com/lightbend/lagom/internal/testkit/ProducerStubFactory.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2009-2017 Lightbend Inc. <https://www.lightbend.com>
+ */
+package com.lightbend.lagom.internal.testkit
```

Comment:
About code reuse:
Code looks very similar but almost all internal classes are stubs implementing public APIs (`Subscriber` and `Topic`) which differ from javadsl to scaladsl already. I think only `TopicBufferActor` could be reused. :-(